props/prop_base.h

```diff
@@ -563,6 +557,16 @@ class PropBase : CommandParser, Looper, protected SaberBase {
     return ret;
   }
 
+  void PrintAndSpeakId() {
```

Comment:
Make this return current_id, then make FindBladeAgain() call this function, that way we don't have to call id() twice, and we don't need to pepper the code with PrintAndSpeakId() calls.

Reply:
the trick is that FindBladeAgain runs FindBestConfig, which needs to retrieve the id().
If that gets it from PrintAndSpeak() as current_id instead, then we;re back to the original issue where the console is spammed with printed (or worse, also spoken) BLADE ID: every loop.
So I'm not sure why we'd return current_id in PrintAndSpeakId(). Future flexibility?

How does it look now?

Reply:
oh wait. Just realized I'm hearing 2 boot.wavs now.
It's caused by the call to SetPreset() in FindBlade(). It was false (announce) before I made it true to play font.wav when Blade ID changed.
Why would SetPreset be playing boot.wav on top of actual boot?
I guess that's why it was false announce originally?
It seems like a weird thing that SetPreset would play boot.wav, and not just font.wav on preset change (or blade ID change as was the intent here).
Adding a delay "fixes" it where only one boot.wav plays, but I feel like that's a hack-patch.
```
    if (millis() < 3000) {
      SetPreset(0, false);
    } else {
      SetPreset(0, true);
    }
```

There's this from hybrid_font maybe why boot.wav is playing from blade ID routines, but nothing's calling BladeDetect in FindBladeAgain()...or is there?
```
  void SB_BladeDetect(bool detected) {
    Effect &X(detected ? SFX_bladein : SFX_bladeout);
    if (X) {
      PlayPolyphonic(&X);
      return;
    }
    if (detected && SFX_boot) {
      PlayPolyphonic(&SFX_boot);
      return;
    }
    beeper.Beep(0.05, 2000.0);
  }
  ```

Reply:
Note - it wasn't 2 boot.wavs. It was font.wav over boot.wav because of the edit to SetPreset(0, true) all the time.
Resolved by having it call SetPreset with the announce variable instead of the delay hack-workaround.